airflow/dags/oncokb_curation.py

```diff
@@ -0,0 +1,136 @@
+            df = pd.read_csv(f,sep='\t')
+        return df
+
+    @task
+    def get_statistic_file():
+        test_name="msk_impact_2017"
+        file_path = local+test_name
+
+        # get the mutation data
+        sample = pd.read_csv(file_path+"/data_clinical_sample.txt", usecols=['SAMPLE_ID', 'CANCER_TYPE','CANCER_TYPE_DETAILED'], delimiter='\t', skiprows=4)
+        mutation = pd.read_csv(file_path+"/data_mutations.txt", usecols=['Hugo_Symbol', 'Tumor_Sample_Barcode','HGVSp_Short'], delimiter='\t',skiprows=1)
+        data = pd.merge(sample, mutation, left_on='SAMPLE_ID', right_on='Tumor_Sample_Barcode').drop(columns=['Tumor_Sample_Barcode'])
+        if filter_path:
+            data = exclude_samples(data,filter_path)
+
+        #all cancer frequency
+        all_number = data[['Hugo_Symbol', 'HGVSp_Short']].value_counts().reset_index(name='Number')
+        all_percentage = data[['Hugo_Symbol', 'HGVSp_Short']].value_counts(normalize=True).reset_index(name='Percentage')
+        all_frequency = pd.merge(all_number, all_percentage, on=['Hugo_Symbol', 'HGVSp_Short'])
+        all_path = base / f"{test_name}_all_frequency.txt"
+        upload_to_s3(all_path,all_frequency)
+        logging.info("upload all_frequecny file")
+
+        #cancer type frequency
+        type_number = data.groupby('CANCER_TYPE')[['Hugo_Symbol', 'HGVSp_Short']].value_counts().rename("Number").reset_index()
+        type_percentage = data.groupby('CANCER_TYPE')[['Hugo_Symbol', 'HGVSp_Short']].value_counts(normalize=True).rename("Percentage")
+        type_frequency = pd.merge(type_number, type_percentage, on=['CANCER_TYPE', 'Hugo_Symbol', 'HGVSp_Short'])
+        type_path = base / f"{test_name}_type_frequency.txt"
+        upload_to_s3(type_path,type_frequency)
+        logging.info("upload type_frequecny file")
+
+        #cancer type detailed frequency
+        detailed_number = data.groupby(['CANCER_TYPE','CANCER_TYPE_DETAILED'])[['Hugo_Symbol', 'HGVSp_Short']].value_counts().rename("Number").reset_index()
+        detailed_percentage = data.groupby(['CANCER_TYPE','CANCER_TYPE_DETAILED'])[['Hugo_Symbol', 'HGVSp_Short']].value_counts(normalize=True).rename("Percentage")
+        detailed_frequency = pd.merge(detailed_number, detailed_percentage, on=['CANCER_TYPE', 'CANCER_TYPE_DETAILED', 'Hugo_Symbol', 'HGVSp_Short'])
+        detailed_path = base / f"{test_name}_detailed_frequency.txt"
+        upload_to_s3(detailed_path,detailed_frequency)
+        logging.info("upload detrailed_frequecny file")
+        
+        return test_name
+
+    @task
+    def merge_mutation(test_name):
+        #Get file from s3
+        all_path = base / f"{test_name}_all_frequency.txt"
+        all_frequency = read_from_s3(all_path)
+        type_path = base / f"{test_name}_type_frequency.txt"
+        type_frequency = read_from_s3(type_path)
+        detailed_path = base / f"{test_name}_detailed_frequency.txt"
+        detailed_frequency = read_from_s3(detailed_path)
```

Comment:
Looks like you always include base with every call to `read_from_s3`.
This a reason why you couldn't include the `base / detailed_path` as part of the code in the `read_from_s3` function.